proxyfilters/connectports.go

```diff
@@ -0,0 +1,41 @@
+package proxyfilters
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/getlantern/proxy/filters"
+)
+
+// RestrictConnectPorts restricts CONNECT requests to the given list of allowed
+// ports and returns either a 400 error if the request is missing a port or a
+// 403 error if the port is not allowed.
+func RestrictConnectPorts(allowedPorts []int) filters.Filter {
```

Comment:
Yes, the proxy package now handles the CONNECT logic - https://github.com/getlantern/proxy/pull/11